LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan/SpreadingFactor.cs

```diff
@@ -13,6 +13,7 @@ namespace LoRaWan
     /// </summary>
     public enum SpreadingFactor
     {
+        UNDEFINED = -1, // This is the value to be used in Basics Station configuration for skipping a DataRate setting
```

Comment:
I find that this definition of SF must remain true to LoRa. The -1 seems to be a configuration thing.

Reply:
Agree. I decided to adapt all your suggestions with 83752834